data-prepper-plugins/aws-sqs-common/src/main/java/org/opensearch/dataprepper/plugins/aws/sqs/common/ClientFactory.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.aws.sqs.common;
+
+import org.opensearch.dataprepper.aws.api.AwsCredentialsOptions;
+import org.opensearch.dataprepper.aws.api.AwsCredentialsSupplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sqs.SqsClient;
+
+import java.util.Map;
+
+/**
+ * A Factory for creating and fetching the <code>SqsClient</code>.
+ */
+public class ClientFactory {
+    private static final Logger LOG = LoggerFactory.getLogger(ClientFactory.class);
+
+    private ClientFactory(){
```

Comment:
Is the constructor required here?

Reply:
As the ClientFactory class consist of all static methods, SonarQuebe suggested adding the private constructor to avoid object creation for static access.